etest_test/profile_test.py

```diff
@@ -0,0 +1,150 @@
+"""Tests for etest.profile."""
+
+from typing import Any, Dict
+
+import pytest
+from hypothesis import assume, given
+from hypothesis import strategies as st
+from pytest_mock import MockerFixture
+
+import etest.profile as sut
+
+
+_valid_architectures = ["amd64", "x86", "armv5", "armv6", "armv7", "arm64", "ppc64"]
+_valid_systemd_architectures = ["amd64", "x86", "arm64", "ppc64"]
+_libc_value = ["glibc", "musl", "uclibc"]
+
+
+@st.composite
+def non_base_profiles(draw: Any) -> Dict[str, Any]:
+    """Strategy for generating non-base profiles."""
+    # Draw values
+    profile = {}
+    profile["libc"] = draw(st.sampled_from(_libc_value))
+    profile["hardened"] = draw(st.booleans())
+    profile["multilib"] = draw(st.booleans())
+    profile["systemd"] = draw(st.booleans())
+
+    # Ensure it isn't a base profile
+    assume(profile != {"libc": "glibc", "hardened": False, "multilib": True, "systemd": False})
+
+    return profile
+
+
+@pytest.mark.parametrize("arch", _valid_architectures)
+def test_base_arch(arch: str) -> None:
+    """Ensure the base profiles work."""
+    result = sut.Profile(False, arch, "glibc", False, True, False)
+
+    # Check profile strings
+    assert result.profile == arch
+
+
+@pytest.mark.parametrize("libc", ["musl", "uclibc"])
+@pytest.mark.parametrize("hardened", [True, False])
+def test_amd64_libc(libc: str, hardened: bool) -> None:
+    """Ensure alternative libcs work on amd64."""
+    arch = "amd64"
+
+    result = sut.Profile(True, arch, libc, hardened, True, False)
+
+    # Check the profile string
+    result_suffix = "hardened" if hardened else "vanilla"
+    assert result.profile == f"{arch}-{libc}-{result_suffix}"
+
+
+@pytest.mark.parametrize("arch", ["armv5", "armv6", "armv7"])
+@given(profile=non_base_profiles())
+def test_invalid_arm_profiles(arch: str, profile: Dict[str, Any]) -> None:
+    """Ensure non-base ARM profiles raise exceptions."""
+    with pytest.raises(sut.InvalidProfileError):
+        sut.Profile(True, arch, profile["libc"], profile["hardened"], profile["multilib"], profile["systemd"])
+
+
+@given(profile=non_base_profiles())
+def test_invalid_arm64_profiles(profile: Dict[str, Any]) -> None:
```

Comment:
We should create a strategy that returns Profiles.  Otherwise, why not have strategies for the properties and encode our base profile idea in one location?